src/util/log.h

```diff
@@ -12,6 +16,22 @@ static const u32 ERROR   = 1 << 3;
 static const u32 ALL     =(1 << 4) - 1; // update this if changing log levels
 }
 
+struct LogMessage {
+    const char *file;
+    u32   line;
+    const char *func;
+    char  message[LOG_BUFFER_SIZE] = {};
+};
+
+struct Logger {
+    u32 log_levels      = LogLevel::ALL;
+    u32 log_levels_file = LogLevel::ALL;
+    FILE *file          = nullptr;
+    std::vector<LogMessage> messages;
```

Comment:
Won't you need a lock on this? Maybe wrap this in a method so you can lock it when messages are added to it.

If you don't lock, the audio and network threads will have trubble logging.

Reply:
Seems likely, yes.